feat: implemented pokemons overview page

zrnkrystnlee opened the pull request:
- added connector and vm for overview of pokemons
- added a card widget where it displays pokemons in GridView
- updated constant file for the different magic numbers, strings, and others
- removed default(0) in stats.dart
- deleted Home page
- tested flutter analyze and dart format

results of flutter analyze and dart format
![image](https://user-images.githubusercontent.com/72370369/227433964-c62c7221-1515-41c6-adab-099dd015b5b5.png)

results of application working
![image](https://user-images.githubusercontent.com/72370369/227434038-2ed739fa-d1cc-456f-9ebf-22cf18c296c4.png)

zrnkrystnlee committed:
feat: implemented pokemons overview page
- added connector and vm for overview of pokemons
- added a card widget where it displays pokemons in GridView
- updated constant file for the different magic numbers, strings, and others
- removed default(0) in stats.dart
- deleted Home page
- tested flutter analyze and dart format

zrnkrystnlee committed:
feat: implemented pokemon details actions
(pr feedback)
- used regex to extract the index from the URL
- added a string_extension for the Pokemon's name
- updated the link of the image in constant

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,34 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text(pokemonOverviewTitle),
```
no need for extra trailing comma 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,34 @@
+          crossAxisCount: defaultTwo,
+          childAspectRatio: defaultOne,
+        ),
+        itemBuilder: (context, index) {
```
kindly rename `context` to `_` since this is unused 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,34 @@
+        ),
+        itemBuilder: (context, index) {
+          final pokemon = pokemons[index];
+          final String imageURL = imageUrlPrefix.replaceAll(RegExp(r'\[index\]'), '${index + 1}') + imageUrlSuffix;
```
see jayvee's pr on how she extracted the pokemon's id 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+
+class PokemonCard extends StatelessWidget {
+  const PokemonCard({
+    required this.pokemon,
```
since pokemon model have only 2 properties, why not just pass the whole model inside your pokemon card instead? 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      margin: const EdgeInsets.all(cardDefaultSize),
```
the thing is u named ur constant as `cardDefaultSize`, yet you're using it for margin which is a bit misleading 😝

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+    return Card(
+      margin: const EdgeInsets.all(cardDefaultSize),
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(cardBorderRadius),
```
no need for extra trailing comma 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(cardBorderRadius),
+      ),
+      elevation: cardDefaultSize,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          Container(
+            height: pokemonNameContainerHeight,
```
no need for this constant, immediately place the value here 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+        children: [
+          Container(
+            height: pokemonNameContainerHeight,
+            padding: const EdgeInsets.symmetric(vertical: textPaddingVertical),
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+            child: Text(
+              pokemon.capitalizeName(),
+              style: const TextStyle(
+                fontSize: pokemonNameFontSize,
```
samer here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+              pokemon.capitalizeName(),
+              style: const TextStyle(
+                fontSize: pokemonNameFontSize,
+                fontWeight: textFontWeight,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+          ),
+          //TODO: Will add necessary codes for getting Primary Type of the Pokemon later
+          Container(
+            height: typeContainerHeight,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+          //TODO: Will add necessary codes for getting Primary Type of the Pokemon later
+          Container(
+            height: typeContainerHeight,
+            padding: const EdgeInsets.symmetric(horizontal: typePaddingHorizontal),
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+            height: typeContainerHeight,
+            padding: const EdgeInsets.symmetric(horizontal: typePaddingHorizontal),
+            decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(typeContainerBorderRadius),
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+                const Text(
+                  type,
+                  style: TextStyle(
+                    fontSize: typeTextFontSize,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+                  type,
+                  style: TextStyle(
+                    fontSize: typeTextFontSize,
+                    fontWeight: textFontWeight,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+                  ),
+                ),
+                Container(
+                  height: typePrimaryBoxHeight,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+                ),
+                Container(
+                  height: typePrimaryBoxHeight,
+                  width: typePrimaryBoxWidth,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+                  height: typePrimaryBoxHeight,
+                  width: typePrimaryBoxWidth,
+                  decoration: BoxDecoration(
+                    borderRadius: BorderRadius.circular(typeBoxBorderRadius),
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+                  child: const Text(
+                    typePrimary,
+                    style: TextStyle(
+                      fontSize: typePrimaryTextFontSize,
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widgets/pokemon_card.dart:
```diff
@@ -0,0 +1,94 @@
+          ),
+          Expanded(
+            child: Padding(
+              padding: const EdgeInsets.all(cardDefaultSize),
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/utils/constants.dart:
```diff
@@ -1,5 +1,36 @@
+const FontWeight textFontWeight = FontWeight.bold;
+
+///Pokemon Overview Page Number Constants
+const double defaultOne = 1.0;
```
no need for this, immediately use this based on where it's used 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utils/constants.dart:
```diff
@@ -1,5 +1,36 @@
+
+///Pokemon Overview Page Number Constants
+const double defaultOne = 1.0;
+const int defaultTwo = 2;
```
same here

shawn-ffufm commented on pokedex_async_redux/lib/utils/constants.dart:
```diff
@@ -1,5 +1,36 @@
+const double defaultOne = 1.0;
+const int defaultTwo = 2;
+const double cardDefaultSize = 5.0;
+const double cardBorderRadius = 15.0;
```
kindly remove unnecessary constants declaration in this dart file 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utils/constants.dart:
```diff
@@ -1,5 +1,36 @@
+const double pokemonImageSize = 150.0;
+
+/// //Pokemon Overview Page Color Constants
+const Color lightColor = Colors.white;
```
u can create a new dart file for these color properties, u can name it as colors.dart 🙂

shawn-ffufm commented on pokedex_async_redux/lib/utils/string_extensions.dart:
```diff
@@ -0,0 +1,5 @@
+extension StringExtension on String {
+  String capitalizeName() {
```
kindly read conversation [here](https://ffufmanilabootcamp.slack.com/archives/C027M886THS/p1679636361031749?thread_ts=1679635559.296549&cid=C027M886THS) 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

zrnkrystnlee committed:
feat: implemented pokemon details actions
(pr feedback)
- used string manipulation instead of regex
- updated constants file- removed unused constants and fixed naming convention
- added getting of pokemon ID and URL of each pokemon in string_extensions.dart
- removed uneccessary extra trailing comas
- renamed context to _ in itemBuilder

zrnkrystnlee committed:
feat: implemented pokemon details actions
(pr feedback)
- removed some constants that can be immediately use where its based
- removed unecessary constants declaration
- added dartx in pubspec.yaml

zrnkrystnlee committed:
feat: implemented pokemon details actions
(pr feedback)
- moved color constants to a separate dart file
